### PR TITLE
fix(core): digest backwards compatibility

### PR DESCRIPTION
**Pull request by Archento**
Reverting back to the old way of schema generation to be compatible with current agents.

We will introduce a transitory period of supporting both schemes in the near future to provide a chance for users to properly migrate,

### CODE REVIEW

File: python/tests/test_model.py [hunk 1]
```diff
@@ -25,6 +27,40 @@ class SuperImportantCheck(Model):
 
         self.assertEqual(result, target_digest, "Digest mismatch")
 
+    def test_calculate_nested_digest_backcompat(self):
```
Comment (qati) [inline python/tests/test_model.py:30]: 👍

File: python/src/uagents/protocol.py [hunk 2]
```diff
@@ -315,7 +315,7 @@ def manifest(self) -> Dict[str, Any]:
 
         for schema_digest, model in all_models.items():
             manifest["models"].append(
-                {"digest": schema_digest, "schema": model.model_json_schema()}
+                {"digest": schema_digest, "schema": model.schema()}
```
Comment (qati) [inline python/src/uagents/protocol.py:318]: why the change here? didn't we add model_json_schema so we don't have to change?

Comment (Archento) [reply python/src/uagents/protocol.py]: Apparently there was a missing positional argument meaning I forgot to add the class method decorator. It should work now without the change.

File: python/src/uagents/models.py [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+
     @staticmethod
     def build_schema_digest(model: Union["Model", Type["Model"]]) -> str:
-        schema = model.model_json_schema()
-        digest = (
-            hashlib.sha256(json.dumps(schema, sort_keys=True).encode("utf8"))
-            .digest()
-            .hex()
-        )
+        schema = model.schema_json(indent=None, sort_keys=True)
```
Comment (qati) [inline python/src/uagents/models.py:18]: why change this? model_json_schema was added

Comment (Archento) [reply python/src/uagents/models.py:18]: Because that is a direct method call from the BaseModel and I don't want to rely on my mock methods.
Also we will need to refactor this method again in the near future when we tackle the manifest topic.

**Review by jrriehl — APPROVED** [review]
LGTM

File: python/pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "uagents"
-version = "0.13.1"
+version = "0.13.2"
```
Comment (jrriehl) [inline python/pyproject.toml:3]: You might want to revert this change so that you have something to commit for the release PR from branch `release/v0.13.2`.

Comment (Archento) [reply python/pyproject.toml]: You're right, thanks. I'll create a separate PR for that :)

File: python/src/uagents/__init__.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
 from .agent import Agent, Bureau  # noqa
 from .context import Context  # noqa
-from .models import Model  # noqa
+from .models import Model, Field  # noqa
```
Comment (lrahmani) [inline python/src/uagents/__init__.py:3]: Field v2 are not json serializable anymore
https://docs.pydantic.dev/latest/migration/#changes-to-pydanticfield

Comment (lrahmani) [reply python/src/uagents/__init__.py:3]: This most probabably implies that all of integrations and any other agents using Field would be broken with pydantic v2

File: python/src/uagents/models.py [hunk 6]
```diff
@@ -1,19 +1,26 @@
...
 
 
+# reverting back to pydantic.v1 BaseModel for backwards compatibility
 class Model(BaseModel):
+    @classmethod
+    def model_json_schema(cls) -> str:
+        return cls.schema_json()
+
+    def model_dump_json(self) -> str:
+        return self.json()
```
Comment (lrahmani) [inline python/src/uagents/models.py:14]: Removed the `indent` and `sort_keys` to keep parity with v1
Example https://github.com/fetchai/uAgents/commit/9dc508d173dedc72ff3365692d4aafeb5f3e4de0#diff-93489236d31a4bf06237078577b1d223280256cf088366b24250fc36e62cf284L263

File: python/src/uagents/protocol.py (hunk 2 above)
Comment (lrahmani) [inline python/src/uagents/protocol.py:318]: Restored the correct function as (I assume) it was mistakingly changed here
https://github.com/fetchai/uAgents/commit/110c1c921da040728d8aa5472b85ce29a6a532f4#diff-6043ffbea937f982ee0f219d8d51acc51c66a5d9ca9faef0a33bb26daf227f81L318
`schema` != `json_schema`
https://docs.pydantic.dev/1.10/usage/models/
https://docs.pydantic.dev/latest/migration/#changes-to-pydanticbasemodel

**Review by lrahmani — COMMENTED** [review]
These changes should restore backward compatibility with uagents 0.12.2 for models, envelopes and protocols

File: python/src/uagents/__init__.py (hunk 5 above)
Comment (Archento) [reply python/src/uagents/__init__.py:3]: Fields in v2 are different but should still be serialisable. What this tells me is that we have been using `Field` outside of its intended purpose.
By properly handling and adding metadata to the messages we should be able to avoid the use of Field altogether.
This can of cause happen down the line, it's just something to keep in mind.

**Review by lrahmani — APPROVED** [review]

**Review by lrahmani — APPROVED** [review]
